### ISSUE TITLE
Check udev rules under ./lib/udev/rules.d

### ISSUE THREAD

**Issue by gvolpe** [Dygmalab/Bazecor#370]
**Is your feature request related to a problem? Please describe.**

I am [trying to package](https://github.com/gvolpe/bazecor-nix) the distributed `AppImage` so that it works on [NixOS](https://nixos.org/), but the current [hard-coded file path](https://github.com/Dygmalab/Bazecor/blob/159eed1d37f3fd1fbf5c17023c12bb683b778281/src/main/index.js#L194) on the udev rules check code is problematic. 

**Describe the solution you'd like**

A minimal solution would be to check for the existence of the `10-dygma.rules` file under `lib/udev/rules.d/` in addition to `/etc/udev/rules.d` like [Chrysalis](https://github.com/keyboardio/Chrysalis/blob/master/static/udev/60-kaleidoscope.rules) does. Another option could be to make the file path configurable.

Additionally, it would be great if the necessary udev rules could be on a separate file (as in Chrysalis) instead of being a [string value](https://github.com/Dygmalab/Bazecor/blob/159eed1d37f3fd1fbf5c17023c12bb683b778281/src/main/index.js#L223) pushed via `echo`. That would make it clearer and it would further help the packaging for distros such as NixOS and Guix. 

**Describe alternatives you've considered**

I tried to package the application from source as well (code also on the same repo I linked above), but packaging Electron apps for Nix is an absolute nightmare, and I got blocked on dependency resolution of the `acorn` library on my last attempt.

**Comment by ajmasia** [comment]
The same issue here. The solution proposed by @gvolpe seems interesting and the the most correct!

**Comment by amesgen** [comment]
FYI: I opened a https://github.com/NixOS/nixpkgs/pull/254264 to package bazecor for Nix via the appimage, and one currently has to vendor the udev rules due to the problem described in this issue.

**Comment by gvolpe** [comment]
@amesgen that's awesome! I've parked my work on this until I get the Defy (should arrive in ~2 weeks now). Hopefully their hardware is better than their software 😄

**Comment by voidzero** [comment]
Can this issue please be addressed? I'm writing a Gentoo ebuild that has the same problem, and if the app just looks at `/usr/lib/udev/rules.d/60-dygma.rules` then all is well with the world.
